easybuild/tools/module_naming_scheme/thematic_mns.py

```diff
@@ -0,0 +1,45 @@
+import os
+import re
+
+from easybuild.tools.module_naming_scheme import ModuleNamingScheme
+from easybuild.tools.module_naming_scheme.utilities import det_full_ec_version
+
+class ThematicMNS(ModuleNamingScheme):
```

Comment:
we already have `CategorizedHMNS` (HMNS for Hierarchical MNS), so `CategorizedMNS` would make more sense, but would be close enough to `CategorizedHMNS` to cause confusion...

it's unclear where the term "thematic" comes from though...

maybe `CategorizedModuleNamingScheme` ?